arangod/RocksDBEngine/RocksDBSettingsManager.cpp

```diff
@@ -146,7 +146,6 @@ ResultT<bool> RocksDBSettingsManager::sync(bool force) {
     _tmpBuilder.clear();  // recycle our builder
 
     auto& dbfeature = _engine.server().getFeature<arangodb::DatabaseFeature>();
-    TRI_ASSERT(!_engine.inRecovery());  // just don't
```

Comment:
The assertion failed for me in some of my manual tests.
It can happen if the server starts up, the recovery flag is still set, and then the ApplicationServer shuts down because an exception happens during startup.
Then we technically are not even in recovery, but the assertion still fails.

Reply:
Thanks